registry/storage/validatorstore.go

```diff
@@ -211,8 +224,12 @@ func (c *validatorStore) handleSharesAdded(shares ...*types.SSVShare) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	// Update byValidatorIndex
 	for _, share := range shares {
+		if share == nil {
+			continue
```

Comment:
this should never be allowed
we should make sure to return error in sharesStorage that if a passed share is nil